File: 백준/Gold/1107. 리모컨/리모컨.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

target = int(input())
n = int(input())
cant_used = list(map(int, input().split()))

min_count = abs(100 - target)
for compare in range(1000001):
    compare = str(compare)
    for v in range(len(compare)):
        if int(compare[v]) in cant_used:
            break
        elif v == len(compare) - 1:
            min_count = min(min_count, abs(int(compare) - target) + len(compare))
print(min_count)

File: 백준/Gold/1107. 리모컨/리모컨.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strings"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n int
	var target, minCnt float64
	var cantUsed string
	fmt.Fscanf(reader, "%f\n%d\n", &target, &n)
	cantUsed, _ = reader.ReadString('\n')
	cantUsed = strings.TrimSpace(cantUsed)
	minCnt = math.Abs(100 - target)
	if minCnt != 0 {
		for i := 0; i < 1000001; i++ {
			compareArr := strings.Split(fmt.Sprintf("%d", i), "")
			for v := 0; v < len(compareArr); v++ {
				if strings.Contains(cantUsed, compareArr[v]) {
					break
				} else if v == len(compareArr)-1 {
					minCnt = math.Min(minCnt, math.Abs(float64(i)-target)+float64(len(compareArr)))
				}
			}
		}
	}
	fmt.Fprintf(writer, "%0.f\n", minCnt)
}
    